clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -109,12 +110,82 @@ class CIRPtrStrideOpLowering
 class CIRLoopOpLowering : public mlir::OpConversionPattern<mlir::cir::LoopOp> {
 public:
   using mlir::OpConversionPattern<mlir::cir::LoopOp>::OpConversionPattern;
+  using LoopKind = mlir::cir::LoopOpKind;
+
+  mlir::LogicalResult
+  fetchCondRegionYields(mlir::Region &condRegion,
+                        mlir::cir::YieldOp &yieldToBody,
+                        mlir::cir::YieldOp &yieldToCont) const {
+    for (auto &bb : condRegion) {
+      if (auto yieldOp = dyn_cast<mlir::cir::YieldOp>(bb.getTerminator())) {
+        if (!yieldOp.getKind().has_value())
+          yieldToCont = yieldOp;
+        else if (yieldOp.getKind() == mlir::cir::YieldOpKind::Continue)
+          yieldToBody = yieldOp;
+        else
+          return mlir::failure();
+      }
+    }
+    return mlir::success();
+  }
+
+  mlir::LogicalResult
+  rewriteWhileLoop(mlir::cir::LoopOp loopOp, OpAdaptor adaptor,
+                   mlir::ConversionPatternRewriter &rewriter) const {
+    auto *currentBlock = rewriter.getInsertionBlock();
+    auto *continueBlock =
+        rewriter.splitBlock(currentBlock, rewriter.getInsertionPoint());
+
+    // Fetch required info from the condition region.
+    auto &condRegion = loopOp.getCond();
+    auto &condFrontBlock = condRegion.front();
+    mlir::cir::YieldOp yieldToBody, yieldToCont;
+    if (fetchCondRegionYields(condRegion, yieldToBody, yieldToCont).failed())
```

Comment:
It's actually not handled, though I think it should not happen. I'll add a check in `fetchCondRegionYields` to ensure if fails if one of the values is not set.

I didn't intend to tackle many corner cases here because I'm working on a refactor for the condition region where it would yield a boolean instead of having this set of basic blocs to manage the control flow with just `cir.yield` and `cir.yield continue`.